core/domain/learner_group_services.py

```diff
@@ -0,0 +1,496 @@
+            learner group syllabus.
+
+    Returns:
+        LearnerGroup. The domain object of the newly created learner group.
+    """
+
+    learner_group_model = learner_group_models.LearnerGroupModel(
+        id=group_id,
+        title=title,
+        description=description,
+        facilitator_user_ids=facilitator_user_ids,
+        student_user_ids=student_ids,
+        invited_student_user_ids=invited_student_ids,
+        subtopic_page_ids=subtopic_page_ids,
+        story_ids=story_ids
+    )
+
+    learner_group_model.update_timestamps()
+    learner_group_model.put()
+
+    learner_group = learner_group_domain.LearnerGroup(
+        learner_group_model.id,
+        learner_group_model.title,
+        learner_group_model.description,
+        learner_group_model.facilitator_user_ids,
+        learner_group_model.student_user_ids,
+        learner_group_model.invited_student_user_ids,
+        learner_group_model.subtopic_page_ids,
+        learner_group_model.story_ids
+    )
+
+    if len(learner_group.invited_student_user_ids) > 0:
+        invite_students_to_learner_group(
+            group_id, learner_group.invited_student_user_ids)
+
+    return learner_group
+
+
+def update_learner_group(
+        group_id, title, description,
+        facilitator_user_ids, student_ids, invited_student_ids,
+        subtopic_page_ids, story_ids
+    ) -> learner_group_domain.LearnerGroup:
+    """Updates a learner group if it is present.
+
+    Args:
+        group_id: str. The id of the learner group to be updated.
+        title: str. The title of the learner group.
+        description: str. The description of the learner group.
+        facilitator_user_ids: str. List of user ids of the facilitators of the
+            learner group.
+        student_ids: list(str). List of user ids of the students of the
+            learner group.
+        invited_student_ids: list(str). List of user ids of the students who
+            have been invited to join the learner group.
+        subtopic_page_ids: list(str). The ids of the subtopics pages that are
+            part of the learner group syllabus. Each subtopic page id is
+            represented as a topicId:subtopicId string.
+        story_ids: list(str). The ids of the stories that are part of the
+            learner group syllabus.
+
+    Returns:
+        learner_group: learner_group_domain.LearnerGroup. The domain object
+        of the updated learner group.
+
+    Raises:
+        Exception. The learner group with the given id does not exist.
+    """
+
+    learner_group_model = learner_group_models.LearnerGroupModel.get_by_id(
+        group_id)
+
+    if not learner_group_model:
+        raise Exception(
+            'The learner group with the given group id does not exist.')
+
+    old_invited_student_ids = set(learner_group_model.invited_student_user_ids)
+    new_invited_student_ids = set(invited_student_ids)
```

Comment:
Aren't there validation checks to make sure these are not duplicated in model itself?

Reply:
Yeah, the validation checks are there but creating them to set is mainly done here to get the difference between the 2 so that we could send new invitations and remove the revoked invitations.